### PR TITLE
Added the D-Place biblio, somewhat overdue

### PR DESCRIPTION
**Pull request by d97hah**

### CODE REVIEW

**Comment by xrotwang** [comment]
@d97hah What version of the D-PLACE bib did you start out with? I'm counting 4,229 refs in the [current bib](https://github.com/D-PLACE/dplace-data/blob/master/sources.bib):
```
(dplace) forkel@shh.mpg.de@dlt4803010l:~/venvs/dplace/dplace-data$ grep "@" sources.bib | wc -l
4229
```
but only 3980 here:
```
(glottolog) forkel@shh.mpg.de@dlt4803010l:~/venvs/glottolog/glottolog$ grep "@" references/bibtex/dplace.bib | wc -l
3980
```

If you did some merging, can you trace back what was merged into what - so I could maybe replace the bib in dplace-data with a link to the one here - and basically move curation of this bib here?

**Comment by xrotwang** [comment]
FWIW here's the missing keys (note that the first couple of entries have been added recently, when we merged the sources of phylogenies into the main bib):
```
Birchall_et_al2016
Bouckaert2012
Bouckaert2018
Bowern2012
Chacon2015
Chang2015
Dunn2011
Greenhill2015
Honkola2013
Hruschka2015
Kitchen2009
Lee2011
Lee2015
Lee_2013
Michael2015
Robinson_2012
Sagart2019
Sicoli2014
Walker2011
Zhang_2019
aberle1961
ames1958
antropovaandkuznetsova1964
aroandshepard1967
atkinson2006
audric1972
ayrout1963
baerandengelbrecht1990
barnett1968
barrett1910
basehart1960
basehart1967
basehart1970
basehart1971
basso1969
basso1970
bassoandopler1971
bastien1985
baumhoff1958
baumhoff1963
beaton1939
best1979a
best1979b
black1973
bohannanandbohannan1966
bois1966
briggs1960
bryant1949
campbell1928
castetterandbell1951
chagnon1971
chagnon1974
chagnon1977
cuentas1929
curtis1926
cushing1896
delaguna1960
dozier1966b
dozier1970
draper1978
drucker1936b
drucker1937b
drucker1965
druckerandheizer1967
duff1951
duttandgeib1987
egganandpandey1979
elmendorf1948
euler1954
evanspritchard1974a
ezell1961
fathauer1954b
fenton1957
forceandforce1972
fortes1949c
fox1967
frachtenberg1921
freed1963
freeman1970
fremerey1978
french1961
garfield1947
gayton1946
gifford1923
gifford1926d
gifford1927
gifford1944
giffordandlowie1928
gluckman1965
goddard1973
goldman1937
goldman1937b
goldschmidtetal1940
goodenough1974
goodwin1937
goodwin1971
grant1925
gray2009
grollemund2015
groslier1958
haile1954
halseath1924
harner1972
harrington1916
harrington1955
hart1954a
hartmann1970
hassig1985
hassig1988
hawley1950
helms1970
heuer1972
hewitt1939
hicks1974
hill1936
hilltout1907
hinz1990
hoben1970
hoben1973
hobson1954
hoebel1935
horton1948
hyde1974
james1960
jenness1955
johnston19551956
jones1914
jorgensen1964
kaut1966
kelly1934
kelly1939
kelly1942
kisteandrynkiewich1976
kluckhohn1946
kniffen1939
kochar1970
kramerandnevermann1938
kroeber1908a
kroeber1908b
kroeber1920
kroeber1945
kroeber1949
kroeberandbarrett1960
kubary1889
kurath1970
lahav1975
lange1959
laski1959
leakey1953
lesser1933
levine1965
linton1945
loeb1932
loeb1933b
malinowski1920
malinowski1926a
malinowski1929a
malinowski1935a
mclendonandoswalt1978
mercer1971
mozino1970
murdock1934c
murdock1965
muriuki1975
murphyandmurphy1974
musil1948
nietschmann1973
nimmo1972
nomlandandkroeber1936
ofaheyandspaulding1974
olson1936b
olson1967
opler1940
opler1940b
opler1943
opler1945
opler1955
opler1969
opler1971a
opler1971b
ortiz1969
park1938b
price1970
price1973
price1984
pringle1970
raswan1934
rattray1960
reiter1938
roberts1951
rockwell1956
rohnerandrohner1970
rosmanandrubel1971
sapir1920
sapir1930
sather1975
schebesta1927a
schebesta1952a
schebestea1952b
schneiderandroberts1956
schwimmer1967
senfft1903a
service1947
service1958
service1963a
shack1974
shipek1968
shternberg1933b
sjoberg1953
smith1899
smithandroberts1954
smole1976
sonnichsen1958
soustelle1956
spier1936
spiro1977
spittel1951
stewart1943
stewart1947
stoll1970
strehlow1913
suttles1955
suttles1958
suttles1960a
suttles1960b
swan1857
titiev1956
trager1966
trigger1987
turnbull1978
underhill1946
underhill1956
vayda1975
voegelinandsteward1954
walker1967
walker1968
wallace1949
wallace1950
ward1948
waterman1910
waterman1925
watsonfranke1974
weiner1976
whiting1985
whitman1947
wijesekera1964
wike1958
zantwijk1990
zigmond1938
```

**Comment by xrotwang** [comment]
@d97hah If you want to, I could merge the missing entries into the bib here.

**Comment by d97hah** [comment]
Yes, why not just merge. The earliest dplace.bib I have is dated 20170518
and has 3980 entries, so what you have is probably something that was
expanded in the meantime (and I don't have that)

Pada tanggal Rab, 8 Apr 2020 pukul 18.01 Robert Forkel <
notifications@github.com> menulis:

**Comment by xrotwang** [comment]
@d97hah Did you intentionally only add **one** related D-PLACE society per reference - even in cases when the ref is used for more than one? E.g.
```
@book{aberle1953,
    author = {Aberle, David F.},
    title = {The Kinship System of the Kalmuk Mongols},
    series = {University of New Mexico Publications in Anthropology},
    number = {8},
    pages = {1-48},
    year = {1953},
    dplace_glottolog_id = {kalm1243},
    dplace_pref_name_for_society = {Kalmyk},
    dplace_xd_id = {xd574},
    key = {Aberle (1953)},
    lgcode = {Kalmyk [xal]}
}
```
when this ref is actually also used for `('xd249', 'Baga')`?

**Comment by xrotwang** [comment]
I guess, the "real" question is whether it makes sense to add the `dplace_*` fields at all, since this will duplicate data which is curated elsewhere (at least currently). This becomes particularly difficult when `lgcode` is hand-curated, i.e. cannot be automatically derived from the `dplace_*` fields, because then we cannot update the file wholesale from a new status in D-PLACE.

So, I think we have two options:

1. Curate the bib here (in Glottolog). This means we should not include the `dplace_*` fields, and `lgcode` will be hand-curated (although one might lookup the current status in D-PLACE periodically).

2. Curate the bib in D-PLACE and overwrite the copy here periodically (augmented with `glottolog_ref_id`) just as we do with `degruyter` or `benjamins`. This would mean `lgcode` would be created from the relations in D-PLACE - but you could still influence these with PRs to D-PLACE.

I'd favour option 2.

**Comment by xrotwang** [comment]
Oh, option 1 is also complicated by the fact that then, each release of D-PLACE which adds references would require a release of Glottolog - which is a weird dependency. It really should be the other way round.

File: languoids/tree/atla1278/nort3146/tend1239/bass1263/bedi1237/bape1239/md.ini [hunk 1]
```diff
@@ -28,7 +28,7 @@ glottolog =
 [hh_ethnologue_comment]
 comment = The Bapen language is (or was) not intelligible to its closest relatives Bassari [bsc] and Bedik [tnr]
 	(**hh:w:SantosFerry:Wey-Pe**:82,
-	**hh:vld:Ferry:Bapen**:353-354). It was nearly extinct in 1971.  I wish to thank Guillaume Segerer for bringing this language to my attention. It is missing from E16/E17/E18/E19/E20/E21/E22/E23. See also: Bedik [tnr], Bassari-Tanda [bsc].
+	**hh:vld:Ferry:Bapen**:353-354). It was nearly extinct in 1971.  I wish to thank Guillaume Segerer for bringing this language to my attention. It is missing from E16/E17/E18/E19/E20/E21/E22/E23. See also: Bassari-Tanda [bsc], Bedik [tnr].
```
Comment (xrotwang) [inline languoids/tree/atla1278/nort3146/tend1239/bass1263/bedi1237/bape1239/md.ini:31]: @d97hah it looks like you should explicitly sort the languages listed after "See also". Otherwise we might get these apparent changes, which makes for confusing histories.

**Comment by d97hah** [comment]
Hi! Yes if D-place is significantly dynamic, it makes sense to
automatically populate the dplace_* fields by a script written by you
(since you are sitting inside dplace). Then the current push can be seen as
its initial population. Yes option 2 is better. One can also influence the
lgcode in a dplace-ref as displayed on Glottolog indirectly by including
the same ref in hh.bib with a specific lgcode set, which is good enough for
me :-)

I initially populated the dplace_* and lgcode fields with one association
by an erroneous assumption. Now fixed in last commit. You take it from here.



Pada tanggal Kam, 9 Apr 2020 pukul 10.28 Robert Forkel <
notifications@github.com> menulis:

**Comment by d97hah** [comment]
Sure, where do you want the commit? Not in the dplace PR? I'll include in
the next one

Pada tanggal Kam, 9 Apr 2020 pukul 10.55 Robert Forkel <
notifications@github.com> menulis:

**Comment by xrotwang** [comment]
@d97hah I'll write a command in `pydplace`, which recreates what you did here from the current `dplace-data` repository. So the workflow for a Glottolog release would be
1. Recreate bib for the latest released D-PLACE version, running `dplace glottologbib` (or similar)
2. Merge this bib into glottolog, possibly updating `glottolog_ref_id` entries, running `glottolog-admin dplacebib`
3. Create `monster.bib`

basically copying what we already have for other bibs, see https://github.com/glottolog/glottolog/blob/master/RELEASING.md#merging-the-bibtex-files

**Comment by d97hah** [comment]
SOunds good! H

Pada tanggal Kam, 9 Apr 2020 pukul 12.06 Robert Forkel <
notifications@github.com> menulis:

**Comment by xrotwang** [comment]
What about this format:
```bibtex
@book{aberle1953,
    author = {Aberle, David F.},
    title = {The Kinship System of the Kalmuk Mongols},
    series = {University of New Mexico Publications in Anthropology},
    number = {8},
    pages = {1-48},
    year = {1953},
    key = {Aberle (1953)},
    lgcode = {Baga [mbul1258], Kalmyk [kalm1243]}
}
```
i.e. `lgcode` has "<D-PLACE prefered name for society> [<D-PLACE> glottocode]". All the `pyglottolog` machinery works with this format, and if you really want the refs related to `hid` (maybe because many of the D-PLACE refs are linked to dialects rather than languages), you could go the way of adding the ref to `hh.bib`?

**Comment by d97hah** [comment]
That works

Pada tanggal Kam, 9 Apr 2020 pukul 15.44 Robert Forkel <
notifications@github.com> menulis:
